### ISSUE TITLE
Add support for slurp'ing a Path

### ISSUE THREAD

**Issue by kennyjwilli** [zcaudate/hara#37]
If you try to `slurp` a `Path` you will receive this exception: 

```
IllegalArgumentException Cannot open <#path:"/home/kenny/compute_software/clake/tasks/deps.edn"> as an InputStream.  clojure.java.io/fn--10946 (io.clj:167)
```

I suggest adding support for `slurp`ing paths by adding the follow to `hara.io.file.path`: 

```clojure
(require '[clojure.java.io :as io] '[hara.io.file :as fs])

(extend Path
  io/IOFactory
  (assoc io/default-streams-impl
    :make-input-stream (fn [path opts] (fs/input-stream path))
    :make-output-stream (fn [path opts] (fs/output-stream path))))
```

I'm happy to submit a PR if you are interested.

**Comment by zcaudate** [comment]
I think it's great idea.

**Comment by zcaudate** [comment]
It looks good. Can the `extend` and subsequent test be moved to the end of their respective files. Then there is no need for the `declare` statement.

**Comment by kennyjwilli** [comment]
Done.

**Comment by zcaudate** [comment]
The changes have been released to v2.8.3 and deployed to [clojars](https://clojars.org/zcaudate/hara.io.file). Thanks for your input.
